mipengine/controller/cleaner.py

```diff
@@ -94,17 +94,19 @@ def release_contextid_for_cleanup(self, context_id: str):
         self._cleanup_file_processor._set_released_true_to_file(context_id=context_id)
 
     def _get_node_info_by_id(self, node_id: str) -> _NodeInfoDTO:
-        global_nodes = self._node_registry.get_all_global_nodes()
-        local_nodes = self._node_registry.get_all_local_nodes()
+        global_nodes = self._node_landscape_aggregator.get_all_global_nodes()
+        local_nodes = self._node_landscape_aggregator.get_all_local_nodes()
 
-        for node in global_nodes + local_nodes:
+        for node in list(global_nodes.values()) + list(local_nodes.values()):
             if node.id == node_id:
                 return _NodeInfoDTO(
                     node_id=node.id,
                     queue_address=":".join([str(node.ip), str(node.port)]),
                     db_address=":".join([str(node.db_ip), str(node.db_port)]),
                     tasks_timeout=controller_config.rabbitmq.celery_tasks_timeout,
                 )
+        else:
+            raise KeyError(f"Node with id '{node_id}' is not currently available.")
```

Comment:
this should just return None if the node_id does exist. An exception should be raised from that part of the code where if the returned object from this call is not a _NodeInfoDTO, causes a problem. Inside method `_get_node_info_by_id` not finding this node_id does actually cause a problem, why should an exception be raised?
`_get_node_info_by_id` is called in line 58
```
node_info = self._get_node_info_by_id(node_id)
```
so the problem would be caused in the next line
```
task_handler = _create_node_task_handler(node_info)
```
I m not completely sure but I think `_create_node_task_handler` (or some call inside that or after that) will anyway raise an exception when `None` is passed as an argument and this whole block(lines 57-68)  are already in a try block so this should be already handled. Were you experiencing something different so you added the raising of the exception?

Reply:
I think its better to raise the exception where the error is.
Else it is hard to identify what went wrong.
I had once a problem that it returned None and it was not clear where the problem was.